battleground-state-changes.html.tmpl

```diff
@@ -105,38 +143,98 @@
         dayjs.extend(dayjs_plugin_relativeTime)
         dayjs.extend(dayjs_plugin_utc)
 
-        $(".timestamp").map(function (idx, tselem) {
-            tselem.innerHTML = "<abbr title='" + tselem.innerHTML + "'>" + dayjs.utc(tselem.innerHTML.replace(" UTC", ""), 'YYYY-MM-DD HH:MM:SS').fromNow() + "</abbr>"
-        })
-
-        const $shrinkButton = $('#shrink_button');
-
-        const shrinkTables = () => {
-            $shrinkButton.html('Expand Tables');
-            $('.table').each(function() { $(this).find('tbody tr:gt(2)').hide(); });
-            localStorage.setItem(shrunkLocalStorageKey, 'true');
+        const prettyTimestamps = () => {
+            $(".timestamp").map(function (idx, tselem) {
+                tselem.innerHTML = "<abbr title='" + tselem.innerHTML + "'>" + dayjs.utc(tselem.innerHTML.replace(" UTC", ""), 'YYYY-MM-DD HH:MM:SS').fromNow() + "</abbr>"
+            })
         };
+        prettyTimestamps();
 
-        const expandTables = () => {
-            $shrinkButton.html('Shrink Tables');
+        const shrinkTablesFeature = 'shrunk';
+        feature(shrinkTablesFeature, '#shrink_button', button => {
+            button.html('Expand Tables');
+            $('.table').each(function() { $(this).find('tbody tr:gt(2)').hide(); });
+        }, button => {
+            button.html('Shrink Tables');
             $('.table').each(function() { $(this).find('tbody tr').show(); });
-            localStorage.removeItem(shrunkLocalStorageKey);
-        };
+        });
+    </script>
 
-        const isShrunk = () => localStorage.getItem(shrunkLocalStorageKey) === 'true';
+    <script type="text/javascript">
+        const liveUpdatesFeature = 'live-updates';
+        feature(liveUpdatesFeature, '#live_update_button', button => {
+            button.html('Disable Live Updates');
+        }, button => {
+            button.html('Enable Live Updates');
+        });
 
-        const shrunkLocalStorageKey = 'shrunk';
+        const notificationFeature = 'notifications';
+        feature(notificationFeature, '#notification_button', button => {
+            button.html('Disable Notifications');
 
-        $shrinkButton.on('click', function() {
-            if (isShrunk()) {
-                expandTables();
-            } else {
-                shrinkTables();
+            if (!('Notification' in window)) {
+                return;
             }
+
+            if (Notification.permission !== "denied") {
+                Notification.requestPermission();
+            }
+        }, button => {
+            button.html('Enable Notifications');
         });
 
-        if (localStorage.getItem(shrunkLocalStorageKey)) {
-            shrinkTables();
-        }
+        (async function() {
+            const fetchLatestCommit = async () => {
+                const response = await fetch("https://api.github.com/repos/alex/nyt-2020-election-scraper/commits?per_page=10", {
```

Comment:
This isn't quite perfect since there are cases where there's a new commit that doesn't update the actual vote totals, just the outputted HTML it'll still get triggered. I'm not sure of a great way of fixing that, though.

Maybe change the commit message that is used when a template is updated vs when the numbers update?

Reply:
unique commit messages sounds like a great idea!